src/librustc/dep_graph/dep_node.rs

```diff
@@ -127,6 +127,11 @@ pub enum DepNode<D: Clone + Debug> {
     TraitItems(D),
     ReprHints(D),
     TraitSelect(Vec<D>),
```

Comment:
Is there a reason we add a `TraitSelectSingle` instead of making `TraitSelect` `SmallVec<[D; 1]>`?